### PR TITLE
Improved the article about session flashes

### PR DESCRIPTION
**Pull request by javiereguiluz**
As discussed here https://github.com/symfony/demo/pull/859  Not sure about the branch though.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Which branch should we merge this in? Thanks!

**Review by nicolas-grekas — APPROVED** [review]
(for 3.4)

File: session/avoid_session_start.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ access the flash messages:
 
 .. code-block:: html+twig
 
-    {% if app.request.hasPreviousSession %}
+    {% if app.request.method == 'POST' %}
```
Comment (xabbuh) [inline session/avoid_session_start.rst]: I do not understand this change. I mean the flash entries will probably be created when processing a POST request, but the reason we register them is that we redirect at the end and we need to display them in response for the subsequent event which will be a GET request then. So with this change the flash messages are likely to never be shown.

Comment (javiereguiluz) [reply session/avoid_session_start.rst]: See https://github.com/symfony/demo/pull/859 for more details. People seem convinced that this is the right fix.

Comment (xabbuh) [reply session/avoid_session_start.rst]: Let me crosspost my comment there too then. :)

**Pull request closed without merging**